### PR TITLE
Change foldM type signature to more closely match foldl

### PR DESCRIPTION
**Pull request by milesfrain**
Related to https://github.com/purescript/purescript-foldable-traversable/pull/111

### CODE REVIEW

**Comment by milesfrain** [comment]
@thomashoneyman Can we merge these PRs and friends as part of 0.14 too?
They're not breaking, so should be uncontroversial.

File: src/Data/List/Lazy.purs [hunk 1]
```diff
@@ -742,12 +742,12 @@ transpose xs =
...
-foldM f a xs =
+foldM :: forall m a b. Monad m => (b -> a -> m b) -> b -> List a -> m b
+foldM f b0 xs =
     case uncons xs of
-         Nothing -> pure a
-         Just { head: b, tail: bs } ->
-                       f a b >>= \a' -> foldM f a' bs
+         Nothing -> pure b0
+         Just { head: a, tail: as } ->
+                       f b0 a >>= \b' -> foldM f b' as
```
Comment (JordanMartinez) [inline src/Data/List/Lazy.purs:750]: Just curious, but why did you use `b0` here?

**Review by JordanMartinez — APPROVED** [review]

File: src/Data/List/Lazy.purs (hunk 1 above)
Comment (thomashoneyman) [reply src/Data/List/Lazy.purs:750]: Yea, can we just do

```purs
foldM f b xs = ...
  Nothing -> pure b
  Just { head: a, tail: as } ->
    f b a >>= \b' -> foldM f b' as
```
```

**Review by hdgarrood — APPROVED** [review]

**Comment by JordanMartinez** [comment]
Thanks @milesfrain!

**Pull request merged**